Add cross platform support

MishaTY opened the pull request:
Requires: https://github.com/CosmosOS/IL2CPU/pull/134

TODO:
 - [x] Documentation
 - [x] Templates (in dotnet command maybe?)
 - [x] Fix plug errors

Also can someone please test on MacOS?

Kiirx commented:
I have a macOS VM on Big Sur
I'm going to try.

Kiirx commented:
Doesnt work with MacOS Big Sur

Kiirx commented on install.sh:
```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
```
Doesnt work with **MacOS Big Sur**

zarlo replied:
do you have bash installed?

Kiirx replied:
bruh yes its just that I picked a random line for the review

zarlo replied:
what is the error you get

Kiirx replied:
> what is the error you get

See the last comment

zarlo commented on install.sh:
```diff
@@ -0,0 +1,71 @@
+
+cp build/HyperV/*.vhdx $InstallDir/Build/HyperV/ &&
+cp build/VMWare/Workstation/* $InstallDir/Build/VMware/Workstation/ &&
+cp build/syslinux/* $InstallDir/Build/ISO/
```
all the && are not needed

MishaTY replied:
They are needed to stop the build when a command fails.

zarlo replied:
you can just put ```set -e``` on the top of the file

Kiirx commented on install.sh:
```diff
@@ -0,0 +1,71 @@
+  printf '\E[31m'; echo "$@"; printf '\E[0m'
+}
+
+if [[ $EUID -ne 0 ]]; then
```
Linux is expecting ]]

zarlo commented on install.sh:
```diff
@@ -0,0 +1,71 @@
+dotnet pack source/Cosmos.Core_Plugs -v:q -nologo &&
+dotnet pack source/Cosmos.HAL2 -v:q -nologo &&
+dotnet pack source/Cosmos.System2 -v:q -nologo &&
+dotnet pack source/Cosmos.System2_Plugs -v:q -nologo &&
```
you may want to set an output path here so it will not change on .net update

Kiirx commented:
Ok I tried with the correct things on macos and I got this: 
![image](https://user-images.githubusercontent.com/43090601/147159251-fee401df-6fc3-4b4a-9463-0db0beca7c0c.png)

MishaTY commented:
Superseded by #2041

Kiirx commented:
the installation with the makefile worked but now how am I supposed to use cosmos at /opt/cosmos/ ?

zarlo commented:
@Kiirx  you add it as a nuget package (a local nuget repo should had of been added)

Kiirx commented:
![Screenshot from 2022-01-13 08-33-27](https://user-images.githubusercontent.com/43090601/149285651-44d17e03-ba20-40a5-b2a5-03e4cc2ee490.png)
I can't find it

Kiirx commented:
```
Build started 1/24/2022 2:37:48 PM.
Logging verbosity is set to: Normal.     1>Project "/media/Datas/dev/SkippleOS/SkippleOS/SkippleOS.csproj" on node 1 (build target(s)).
     1>GenerateTargetFrameworkMonikerAttribute:
       Skipping target "GenerateTargetFrameworkMonikerAttribute" because all output files are up-to-date with respect to the input files.
       CoreGenerateAssemblyInfo:
       Skipping target "CoreGenerateAssemblyInfo" because all output files are up-to-date with respect to the input files.
       CoreCompile:
         /usr/share/dotnet/dotnet exec "/usr/share/dotnet/sdk/6.0.101/Roslyn/bincore/csc.dll" /noconfig /unsafe- /checked- /nowarn:1701,1702,1701,1702 /fullpaths /nostdlib+ /errorreport:prompt /warn:5 /define:TRACE;DEBUG;NET;NET5_0;NETCOREAPP;NET5_0_OR_GREATER;NETCOREAPP1_0_OR_GREATER;NETCOREAPP1_1_OR_GREATER;NETCOREAPP2_0_OR_GREATER;NETCOREAPP2_1_OR_GREATER;NETCOREAPP2_2_OR_GREATER;NETCOREAPP3_0_OR_GREATER;NETCOREAPP3_1_OR_GREATER /highentropyva+ /reference:/home/kiirox/.nuget/packages/cosmos.common/0.1.0-localbuild/lib/net5.0/Cosmos.Common.dll /reference:/home/kiirox/.nuget/packages/cosmos.core/0.1.0-localbuild/lib/net5.0/Cosmos.Core.dll /reference:/home/kiirox/.nuget/packages/cosmos.debug.kernel/0.1.0-localbuild/lib/netstandard2.0/Cosmos.Debug.Kernel.dll /reference:/home/kiirox/.nuget/packages/cosmos.hal2/0.1.0-localbuild/lib/net5.0/Cosmos.HAL2.dll /reference:/home/kiirox/.nuget/packages/cosmos.system2/0.1.0-localbuild/lib/net5.0/Cosmos.System2.dll /reference:/home/kiirox/.nuget/packages/il2cpu.api/0.1.0-localbuild/lib/netstandard2.0/IL2CPU.API.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/Microsoft.CSharp.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/Microsoft.VisualBasic.Core.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/Microsoft.VisualBasic.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/Microsoft.Win32.Primitives.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/mscorlib.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/netstandard.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.AppContext.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Buffers.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Collections.Concurrent.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Collections.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Collections.Immutable.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Collections.NonGeneric.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Collections.Specialized.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.ComponentModel.Annotations.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.ComponentModel.DataAnnotations.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.ComponentModel.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.ComponentModel.EventBasedAsync.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.ComponentModel.Primitives.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.ComponentModel.TypeConverter.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Configuration.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Console.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Core.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Data.Common.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Data.DataSetExtensions.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Data.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Diagnostics.Contracts.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Diagnostics.Debug.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Diagnostics.DiagnosticSource.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Diagnostics.FileVersionInfo.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Diagnostics.Process.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Diagnostics.StackTrace.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Diagnostics.TextWriterTraceListener.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Diagnostics.Tools.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Diagnostics.TraceSource.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Diagnostics.Tracing.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Drawing.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Drawing.Primitives.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Dynamic.Runtime.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Formats.Asn1.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Globalization.Calendars.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Globalization.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Globalization.Extensions.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.IO.Compression.Brotli.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.IO.Compression.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.IO.Compression.FileSystem.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.IO.Compression.ZipFile.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.IO.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.IO.FileSystem.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.IO.FileSystem.DriveInfo.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.IO.FileSystem.Primitives.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.IO.FileSystem.Watcher.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.IO.IsolatedStorage.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.IO.MemoryMappedFiles.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.IO.Pipes.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.IO.UnmanagedMemoryStream.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Linq.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Linq.Expressions.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Linq.Parallel.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Linq.Queryable.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Memory.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Net.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Net.Http.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Net.Http.Json.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Net.HttpListener.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Net.Mail.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Net.NameResolution.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Net.NetworkInformation.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Net.Ping.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Net.Primitives.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Net.Requests.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Net.Security.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Net.ServicePoint.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Net.Sockets.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Net.WebClient.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Net.WebHeaderCollection.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Net.WebProxy.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Net.WebSockets.Client.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Net.WebSockets.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Numerics.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Numerics.Vectors.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.ObjectModel.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Reflection.DispatchProxy.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Reflection.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Reflection.Emit.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Reflection.Emit.ILGeneration.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Reflection.Emit.Lightweight.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Reflection.Extensions.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Reflection.Metadata.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Reflection.Primitives.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Reflection.TypeExtensions.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Resources.Reader.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Resources.ResourceManager.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Resources.Writer.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Runtime.CompilerServices.Unsafe.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Runtime.CompilerServices.VisualC.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Runtime.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Runtime.Extensions.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Runtime.Handles.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Runtime.InteropServices.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Runtime.InteropServices.RuntimeInformation.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Runtime.Intrinsics.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Runtime.Loader.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Runtime.Numerics.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Runtime.Serialization.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Runtime.Serialization.Formatters.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Runtime.Serialization.Json.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Runtime.Serialization.Primitives.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Runtime.Serialization.Xml.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Security.Claims.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Security.Cryptography.Algorithms.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Security.Cryptography.Csp.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Security.Cryptography.Encoding.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Security.Cryptography.Primitives.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Security.Cryptography.X509Certificates.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Security.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Security.Principal.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Security.SecureString.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.ServiceModel.Web.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.ServiceProcess.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Text.Encoding.CodePages.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Text.Encoding.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Text.Encoding.Extensions.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Text.Encodings.Web.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Text.Json.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Text.RegularExpressions.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Threading.Channels.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Threading.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Threading.Overlapped.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Threading.Tasks.Dataflow.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Threading.Tasks.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Threading.Tasks.Extensions.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Threading.Tasks.Parallel.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Threading.Thread.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Threading.ThreadPool.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Threading.Timer.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Transactions.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Transactions.Local.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.ValueTuple.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Web.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Web.HttpUtility.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Windows.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Xml.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Xml.Linq.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Xml.ReaderWriter.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Xml.Serialization.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Xml.XDocument.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Xml.XmlDocument.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Xml.XmlSerializer.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Xml.XPath.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/System.Xml.XPath.XDocument.dll /reference:/usr/share/dotnet/packs/Microsoft.NETCore.App.Ref/5.0.0/ref/net5.0/WindowsBase.dll /debug+ /debug:portable /filealign:512 /optimize- /out:obj/Debug/net5.0/SkippleOS.dll /refout:obj/Debug/net5.0/ref/SkippleOS.dll /target:library /warnaserror- /utf8output /deterministic+ /langversion:9.0 /analyzerconfig:obj/Debug/net5.0/SkippleOS.GeneratedMSBuildEditorConfig.editorconfig /analyzerconfig:/usr/share/dotnet/sdk/6.0.101/Sdks/Microsoft.NET.Sdk/analyzers/build/config/analysislevel_5_default.editorconfig /analyzer:/usr/share/dotnet/sdk/6.0.101/Sdks/Microsoft.NET.Sdk/targets/../analyzers/Microsoft.CodeAnalysis.CSharp.NetAnalyzers.dll /analyzer:/usr/share/dotnet/sdk/6.0.101/Sdks/Microsoft.NET.Sdk/targets/../analyzers/Microsoft.CodeAnalysis.NetAnalyzers.dll Kernel.cs Shell/Cmds/Console/cClear.cs Shell/Cmds/Console/cEcho.cs Shell/Cmds/Console/cHelp.cs Shell/Cmds/Console/cKeyboardMap.cs Shell/Cmds/Console/cWhoAmI.cs Shell/Cmds/File/cCat.cs Shell/Cmds/File/cCD.cs Shell/Cmds/File/cCreateDir.cs Shell/Cmds/File/cCreateFile.cs Shell/Cmds/File/cListDir.cs Shell/Cmds/File/cRemoveDir.cs Shell/Cmds/File/cRemoveFile.cs Shell/Cmds/Power/cReboot.cs Shell/Cmds/Power/cShutdown.cs Shell/ShellInfo.cs Shell/ShellManager.cs "obj/Debug/net5.0/.NETCoreApp,Version=v5.0.AssemblyAttributes.cs" obj/Debug/net5.0/SkippleOS.AssemblyInfo.cs /warnaserror+:NU1605
         CompilerServer: server - server processed compilation - 48ddd588-faa9-4d69-b89d-fd3af4f4f4b6
       CopyFilesToOutputDirectory:
         Copying file from "/media/Datas/dev/SkippleOS/SkippleOS/obj/Debug/net5.0/SkippleOS.dll" to "/media/Datas/dev/SkippleOS/SkippleOS/bin/Debug/net5.0/SkippleOS.dll".
         SkippleOS -> /media/Datas/dev/SkippleOS/SkippleOS/bin/Debug/net5.0/SkippleOS.dll
         Copying file from "/media/Datas/dev/SkippleOS/SkippleOS/obj/Debug/net5.0/SkippleOS.pdb" to "/media/Datas/dev/SkippleOS/SkippleOS/bin/Debug/net5.0/SkippleOS.pdb".
       IL2CPU:
         /opt/cosmos/Build/IL2CPU/IL2CPU KernelPkg:
         EnableDebug:False
         EnableStackCorruptionDetection:True
         StackCorruptionDetectionLevel:MethodFooters
         DebugMode:Source
         TraceAssemblies:
         DebugCom:0
         TargetAssembly:/media/Datas/dev/SkippleOS/SkippleOS/bin/Debug/net5.0/SkippleOS.dll
         OutputFilename:/media/Datas/dev/SkippleOS/SkippleOS/bin/Debug/net5.0/SkippleOS.asm
         EnableLogging:True
         EmitDebugSymbols:True
         IgnoreDebugStubAttribute:False
         CompileVBEMultiboot:False
         VBEResolution:800x600x32
         References:/media/Datas/dev/SkippleOS/SkippleOS/obj/Debug/net5.0/SkippleOS.dll
         References:/home/kiirox/.nuget/packages/cosmos.common/0.1.0-localbuild/lib/net5.0/Cosmos.Common.dll
         References:/home/kiirox/.nuget/packages/cosmos.core/0.1.0-localbuild/lib/net5.0/Cosmos.Core.dll
         References:/home/kiirox/.nuget/packages/cosmos.debug.kernel/0.1.0-localbuild/lib/netstandard2.0/Cosmos.Debug.Kernel.dll
         References:/home/kiirox/.nuget/packages/cosmos.hal2/0.1.0-localbuild/lib/net5.0/Cosmos.HAL2.dll
         References:/home/kiirox/.nuget/packages/cosmos.system2/0.1.0-localbuild/lib/net5.0/Cosmos.System2.dll
         References:/home/kiirox/.nuget/packages/il2cpu.api/0.1.0-localbuild/lib/netstandard2.0/IL2CPU.API.dll
         PlugsReferences:/opt/cosmos/Kernel/Cosmos.Core_Asm.dll
         PlugsReferences:/opt/cosmos/Kernel/Cosmos.Core_Plugs.dll
         PlugsReferences:/opt/cosmos/Kernel/Cosmos.Debug.Kernel.Plugs.Asm.dll
         PlugsReferences:/opt/cosmos/Kernel/Cosmos.System2_Plugs.dll
         userkit: /opt/cosmos/
         Message: Executing IL2CPU on assembly
         Message: Kernel Base: Cosmos.System.Kernel
         Message: Checking target assembly: /media/Datas/dev/SkippleOS/SkippleOS/bin/Debug/net5.0/SkippleOS.dll
     1>Message : warning : Loading plugs from assembly: Cosmos.Core_Asm, Version=1.0.0.0, Culture=neutral, PublicKeyToken=5ae71220097cb983
     1>Message : warning : Loading plugs from assembly: Cosmos.Core_Plugs, Version=1.0.0.0, Culture=neutral, PublicKeyToken=5ae71220097cb983
     1>Message : warning : Loading plugs from assembly: Cosmos.Debug.Kernel.Plugs.Asm, Version=1.0.0.0, Culture=neutral, PublicKeyToken=5ae71220097cb983
     1>Message : warning : Loading plugs from assembly: Cosmos.System2_Plugs, Version=1.0.0.0, Culture=neutral, PublicKeyToken=5ae71220097cb983
     1>Message : warning : Invalid plug method! Target method not found. : SystemStringCosmosCore_PlugsSystemTimeZoneInfoImplTryGetLocalizedNameByMuiNativeResourceSystemString
     1>Message : warning : Invalid plug method! Target method not found. : SystemStringCosmosCore_PlugsSystemGlobalizationCultureInfoImplGetUserDefaultLocaleName
     1>Message : warning : Invalid plug method! Target method not found. : SystemUInt32CosmosCore_PlugsInteropAdvapi32ImplEventUnregisterSystemInt64
     1>Message : warning : Invalid plug method! Target method not found. : SystemInt32CosmosCore_PlugsInteropAdvapi32ImplRegCloseKeySystemIntPtr
     1>Message : warning : Invalid plug method! Target method not found. : SystemInt32CosmosCore_PlugsInteropKernel32ImplGetCurrentProcessId
     1>Message : warning : Invalid plug method! Target method not found. : SystemInt64CosmosCore_PlugsInteropKernel32ImplVerSetConditionMaskSystemUInt64SystemUInt32SystemByte
     1>Message : warning : Invalid plug method! Target method not found. : SystemInt32CosmosCore_PlugsInteropKernel32ImplGetCalendarInfoExSystemStringSystemUInt32SystemIntPtrSystemUInt32SystemIntPtrSystemInt32SystemInt32
     1>Message : warning : Invalid plug method! Target method not found. : SystemUInt32CosmosCore_PlugsInteropKernel32ImplExpandEnvironmentStringsSystemStringSystemCharSystemUInt32
     1>Message : warning : Invalid plug method! Target method not found. : SystemBooleanCosmosSystem_PlugsSystemDateTimeImplSystemSupportsLeapSeconds
     1>Message : warning : Invalid plug method! Target method not found. : SystemInt64CosmosSystem_PlugsSystemDateTimeImplToFileTimeLeapSecondsAwareSystemInt64
     1>Message : warning : Invalid plug method! Target method not found. : SystemVoidCosmosSystem_PlugsSystemThreadingThreadImplStartupSetApartmentStateInternal
     1>Message : warning : Invalid plug method! Target method not found. : SystemVoidCosmosSystem_PlugsSystemIOCosmosFileSystemCreateDirectorySystemStringSystemBytearray
     1>Message : warning : Invalid plug method! Target method not found. : SystemBooleanCosmosSystem_PlugsSystemIOCosmosFileSystemDirectoryExistsSystemString
     1>Message : warning : Invalid plug method! Target method not found. : SystemBooleanCosmosSystem_PlugsSystemIOCosmosFileSystemFileExistsSystemString
     1>Message : warning : Invalid plug method! Target method not found. : SystemBooleanCosmosSystem_PlugsSystemIOFileSystemImplDirectoryExistsSystemStringSystemInt32
     1>Message : warning : Invalid plug method! Target method not found. : SystemInt32CosmosSystem_PlugsSystemGlobalizationCultureDataImplGetLocaleInfoExIntSystemStringSystemUInt32
     1>Message : warning : Invalid plug method! Target method not found. : SystemObjectarrayCosmosSystem_PlugsSystemGlobalizationJapaneseCalendarImplNlsGetJapaneseEras
     1>IL2CPU : error : Exception: System.Exception:
         Native code encountered, plug required.
           DO NOT REPORT THIS AS A BUG.
           Please see http://www.gocosmos.org/docs/plugs/missing/
           Need plug for: System.Int32  Interop+Sys.Stat(&System.Byte, &Interop+Sys+FileStatus)(Plug Signature: System_Int32__Interop_Sys_Stat__System_Byte___Interop_Sys_FileStatus_ ). 
           Static: True
           Assembly: System.IO.FileSystem, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
           Called from:
         Interop+Sys::Int32 Stat(System.ReadOnlySpan`1[System.Char], FileStatus ByRef)
         System.IO.FileSystem::Boolean FileExists(System.ReadOnlySpan`1[System.Char], Int32, ErrorInfo ByRef)
         System.IO.FileSystem::Boolean FileExists(System.ReadOnlySpan`1[System.Char])
         System.IO.File::Boolean Exists(System.String)
         Cosmos.System.FileSystem.CosmosVFS::Cosmos.System.FileSystem.Listing.DirectoryEntry CreateFile(System.String)
         Cosmos.System.FileSystem.VFS.VFSBase::Cosmos.System.FileSystem.Listing.DirectoryEntry CreateFile(System.String)
         
         
            at Cosmos.IL2CPU.ILScanner.ScanMethod(MethodBase aMethod, Boolean aIsPlug, String sourceItem) in /home/kiirox/Documents/Cosmos/IL2CPU/source/Cosmos.IL2CPU/ILScanner.cs:line 624
            at Cosmos.IL2CPU.ILScanner.ScanQueue() in /home/kiirox/Documents/Cosmos/IL2CPU/source/Cosmos.IL2CPU/ILScanner.cs:line 788
            at Cosmos.IL2CPU.ILScanner.Execute(MethodBase aStartMethod, IEnumerable`1 plugsAssemblies) in /home/kiirox/Documents/Cosmos/IL2CPU/source/Cosmos.IL2CPU/ILScanner.cs:line 272
            at Cosmos.IL2CPU.CompilerEngine.Execute() in /home/kiirox/Documents/Cosmos/IL2CPU/source/Cosmos.IL2CPU/CompilerEngine.cs:line 222
         IL2CPU task took 00:00:02.5652488
     1>Done Building Project "/media/Datas/dev/SkippleOS/SkippleOS/SkippleOS.csproj" (build target(s)) -- FAILED.

Build FAILED.

       "/media/Datas/dev/SkippleOS/SkippleOS/SkippleOS.csproj" (build target) (1) ->
       (IL2CPU target) -> 
         Message : warning : Loading plugs from assembly: Cosmos.Core_Asm, Version=1.0.0.0, Culture=neutral, PublicKeyToken=5ae71220097cb983
         Message : warning : Loading plugs from assembly: Cosmos.Core_Plugs, Version=1.0.0.0, Culture=neutral, PublicKeyToken=5ae71220097cb983
         Message : warning : Loading plugs from assembly: Cosmos.Debug.Kernel.Plugs.Asm, Version=1.0.0.0, Culture=neutral, PublicKeyToken=5ae71220097cb983
         Message : warning : Loading plugs from assembly: Cosmos.System2_Plugs, Version=1.0.0.0, Culture=neutral, PublicKeyToken=5ae71220097cb983
         Message : warning : Invalid plug method! Target method not found. : SystemStringCosmosCore_PlugsSystemTimeZoneInfoImplTryGetLocalizedNameByMuiNativeResourceSystemString
         Message : warning : Invalid plug method! Target method not found. : SystemStringCosmosCore_PlugsSystemGlobalizationCultureInfoImplGetUserDefaultLocaleName
         Message : warning : Invalid plug method! Target method not found. : SystemUInt32CosmosCore_PlugsInteropAdvapi32ImplEventUnregisterSystemInt64
         Message : warning : Invalid plug method! Target method not found. : SystemInt32CosmosCore_PlugsInteropAdvapi32ImplRegCloseKeySystemIntPtr
         Message : warning : Invalid plug method! Target method not found. : SystemInt32CosmosCore_PlugsInteropKernel32ImplGetCurrentProcessId
         Message : warning : Invalid plug method! Target method not found. : SystemInt64CosmosCore_PlugsInteropKernel32ImplVerSetConditionMaskSystemUInt64SystemUInt32SystemByte
         Message : warning : Invalid plug method! Target method not found. : SystemInt32CosmosCore_PlugsInteropKernel32ImplGetCalendarInfoExSystemStringSystemUInt32SystemIntPtrSystemUInt32SystemIntPtrSystemInt32SystemInt32
         Message : warning : Invalid plug method! Target method not found. : SystemUInt32CosmosCore_PlugsInteropKernel32ImplExpandEnvironmentStringsSystemStringSystemCharSystemUInt32
         Message : warning : Invalid plug method! Target method not found. : SystemBooleanCosmosSystem_PlugsSystemDateTimeImplSystemSupportsLeapSeconds
         Message : warning : Invalid plug method! Target method not found. : SystemInt64CosmosSystem_PlugsSystemDateTimeImplToFileTimeLeapSecondsAwareSystemInt64
         Message : warning : Invalid plug method! Target method not found. : SystemVoidCosmosSystem_PlugsSystemThreadingThreadImplStartupSetApartmentStateInternal
         Message : warning : Invalid plug method! Target method not found. : SystemVoidCosmosSystem_PlugsSystemIOCosmosFileSystemCreateDirectorySystemStringSystemBytearray
         Message : warning : Invalid plug method! Target method not found. : SystemBooleanCosmosSystem_PlugsSystemIOCosmosFileSystemDirectoryExistsSystemString
         Message : warning : Invalid plug method! Target method not found. : SystemBooleanCosmosSystem_PlugsSystemIOCosmosFileSystemFileExistsSystemString
         Message : warning : Invalid plug method! Target method not found. : SystemBooleanCosmosSystem_PlugsSystemIOFileSystemImplDirectoryExistsSystemStringSystemInt32
         Message : warning : Invalid plug method! Target method not found. : SystemInt32CosmosSystem_PlugsSystemGlobalizationCultureDataImplGetLocaleInfoExIntSystemStringSystemUInt32
         Message : warning : Invalid plug method! Target method not found. : SystemObjectarrayCosmosSystem_PlugsSystemGlobalizationJapaneseCalendarImplNlsGetJapaneseEras


       "/media/Datas/dev/SkippleOS/SkippleOS/SkippleOS.csproj" (build target) (1) ->
       (IL2CPU target) -> 
         IL2CPU : error : Exception: System.Exception:

    21 Warning(s)
    1 Error(s)

Time Elapsed 00:00:04.96
```
When building with un empty error

Kiirx commented:
```
/home/kiirox/Documents/Cosmos/Directory.Build.props(40,5): error MSB4019: The imported project "/home/kiirox/Documents/Cosmos/build/Targets/Import.props" was not found. Confirm that the expression in the Import declaration "build/Targets/Import.props" is correct, and that the file exists on disk. [/home/kiirox/Documents/Cosmos/source/Cosmos.Debug.Kernel/Cosmos.Debug.Kernel.csproj]
```
![image](https://user-images.githubusercontent.com/43090601/154869542-d36c11e2-27d0-4161-a8b9-d25514716cb0.png)
![image](https://user-images.githubusercontent.com/43090601/154869572-4effd148-41a8-4b5e-8b5e-dacab36cde97.png)

Kiirx commented:
```
0>IL2CPU: Error  : Exception: System.Exception:
0>Native code encountered, plug required.
0>  DO NOT REPORT THIS AS A BUG.
0>  Please see http://www.gocosmos.org/docs/plugs/missing/
0>  Need plug for: System.Int32  Interop+Sys.Stat(&System.Byte, &Interop+Sys+FileStatus)(Plug Signature: System_Int32__Interop_Sys_Stat__System_Byte___Interop_Sys_FileStatus_ ). 
0>  Static: True
0>  Assembly: System.IO.FileSystem, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
0>  Called from:
0>Interop+Sys::Int32 Stat(System.ReadOnlySpan`1[System.Char], FileStatus ByRef)
0>System.IO.FileSystem::Boolean FileExists(System.ReadOnlySpan`1[System.Char], Int32, ErrorInfo ByRef)
0>System.IO.FileSystem::Boolean FileExists(System.ReadOnlySpan`1[System.Char])
0>System.IO.File::Boolean Exists(System.String)
0>Cosmos.System.FileSystem.CosmosVFS::Cosmos.System.FileSystem.Listing.DirectoryEntry CreateFile(System.String)
0>Cosmos.System.FileSystem.VFS.VFSBase::Cosmos.System.FileSystem.Listing.DirectoryEntry CreateFile(System.String)
```
error when building

MishaTY commented:
> ```
> /home/kiirox/Documents/Cosmos/Directory.Build.props(40,5): error MSB4019: The imported project "/home/kiirox/Documents/Cosmos/build/Targets/Import.props" was not found. Confirm that the expression in the Import declaration "build/Targets/Import.props" is correct, and that the file exists on disk. [/home/kiirox/Documents/Cosmos/source/Cosmos.Debug.Kernel/Cosmos.Debug.Kernel.csproj]
> ```
> 
> ![image](https://user-images.githubusercontent.com/43090601/154869542-d36c11e2-27d0-4161-a8b9-d25514716cb0.png) ![image](https://user-images.githubusercontent.com/43090601/154869572-4effd148-41a8-4b5e-8b5e-dacab36cde97.png)
> 
> (edit: fixed by renaming Build folder to build)

fixed it

Kiirx commented on Makefile:
```diff
@@ -0,0 +1,104 @@
+
+IL2CPU_BRANCH = crossplatform
+XSHARP_BRANCH = master
+COMMON_BRANCH = crossplatform
```
Should be changed to master since it got merged

Kiirx replied:
for common branch

Kiirx commented on Makefile:
```diff
@@ -0,0 +1,104 @@
+	@cp -r $(THISDIR)/source/Cosmos.HAL2/bin/Debug/net5.0/publish/*.dll $(DESTDIR)/Kernel/
+	@cp -r $(THISDIR)/source/Cosmos.Debug.Kernel.Plugs.Asm/bin/Debug/netstandard2.0/publish/*.dll $(DESTDIR)/Kernel/
+
+	@cp -r $(THISDIR)/build/HyperV/*.vhdx $(DESTDIR)/Build/HyperV/
+	@cp -r $(THISDIR)/build/VMWare/Workstation/* $(DESTDIR)/Build/VMware/Workstation/
+	@cp -r $(THISDIR)/build/syslinux/* $(DESTDIR)/Build/ISO/
```
Fix `build` to `Build`

Kiirx replied:
yasm is in build folder but the make file is confused so it uses build but it should use Build folder instead

zarlo commented on source/Cosmos.Build.Tasks/build/Cosmos.Build.targets:
```diff
@@ -27,8 +27,10 @@
-        <MkisofsToolPath Condition = "'$(MkisofsToolPath)' == ''">$(CosmosToolsPath)mkisofs\win\</MkisofsToolPath>
-        <MkisofsToolExe Condition = "'$(MkisofsToolExe)' == ''">mkisofs.exe</MkisofsToolExe>
+        <MkisofsToolPath Condition = "'$(MkisofsToolPath)' == '' AND '$(OS)' == 'Windows_NT'">$(CosmosToolsPath)mkisofs\win\</MkisofsToolPath>
+        <MkisofsToolPath Condition = "'$(MkisofsToolPath)' == '' AND '$(OS)' != 'Windows_NT'">/usr/bin/</MkisofsToolPath>
```
it might be better to not set the path and use PATH search find it

Kiirx commented:
Ok now we need to do VFS plug and it should be done, but I don't know how to do it haha its confusing

quajak commented on Build/Tools/Yasm/LICENSE.txt:
```diff
@@ -0,0 +1,74 @@
+Redistribution and use in source and binary forms, with or without
```
I dont think we should ship two versions of yasm with cosmos. We currently have a version at [https://github.com/CosmosOS/Cosmos/tree/master/source/Cosmos.Build.Tasks/tools/yasm/win](https://github.com/CosmosOS/Cosmos/tree/master/source/Cosmos.Build.Tasks/tools/yasm/win) . Can we add a step to the build process to copy it from there to Build/Tools/Yasm ?

quajak reviewed (changes requested):
Otherwise it looks fine

quajak commented:
Could we also add a few lines to the documentation somewhere on how to use Cosmos on linux?

MishaTY commented:
> Could we also add a few lines to the documentation somewhere on how to use Cosmos on linux?

Done

quajak commented on appveyor.yml:
```diff
@@ -8,7 +8,7 @@ clone_folder: c:\Cosmos
 init:
 - cmd: |
-    git clone https://github.com/CosmosOS/IL2CPU.git c:\IL2CPU --depth 1
+    git clone https://github.com/CosmosOS/IL2CPU.git c:\IL2CPU --depth 1 --branch crossplatform 
```
we should configure to run tests on both linux and windows but for now just run the windows tests

MishaTY replied:
@valentinbreiz make sure to revert this change

quajak commented on source/Cosmos.Debug.HyperVServer/Cosmos.Debug.HyperVServer.csproj:
```diff
@@ -1,7 +1,7 @@
 
     <PropertyGroup>
-        <TargetFramework>net48</TargetFramework>
+        <TargetFramework>net5.0</TargetFramework>
```
Does this work under windows still? IIRC there was some reason to keep it on net48

zarlo replied:
we can even use hyperV on linux so... if it does break it would be safe to revert and just add build conditions to not include this on linux

Kiirx commented:
I think that making a linux package for cosmos would be great, I can do research about it

zarlo commented:
@Kiirx there are a lot of package formats if i was you i would just add it the the AUR (https://aur.archlinux.org/) and the MPR (https://mpr.makedeb.org/) and let every ones else fend for themself
